File: app/src/main/java/za/co/kernelpanic/cloudy/data/Forecast.java
package za.co.kernelpanic.cloudy.data;


import com.google.gson.annotations.SerializedName;

import java.util.List;


public class Forecast {

    @SerializedName("temp")
    private Temperature temperature;
    @SerializedName("weather")
    private List<Weather> weather;
    @SerializedName("pressure")
    private double pressure;
    @SerializedName("humidity")
    private double humidity;
    @SerializedName("speed")
    private double windSpeed;
    @SerializedName("deg")
    private double windDirection;

    public double getPressure() {
        return pressure;
    }

    public double getHumidity() {
        return humidity;
    }

    public Temperature getTemperature() {
        return temperature;
    }

    public List<Weather> getWeather() {
        return weather;
    }

    public double getWindSpeed() {
        return windSpeed;
    }

    public double getWindDirection() {
        return windDirection;
    }


}

File: app/src/main/java/za/co/kernelpanic/cloudy/data/Forecast.kt
package za.co.kernelpanic.cloudy.data

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass


@JsonClass (generateAdapter = true)
data class Forecast (@Json(name = "temp" )val temperature: Temperature,
                     val weather : List<Weather>,
                     val pressure: Double,
                     val humidity: Double,
                     @Json(name = "speed") val windSpeed: Double,
                     @Json(name = "deg") val windDirection: Double)